.circleci/config.yml

```diff
@@ -6,53 +6,47 @@ jobs:
       - save_cache:
           key: dependency-cache-{{ checksum "Gopkg.lock" }}
           paths:
             - vendor
-      - run:
-          name: Linting
-          command: golint -set_exit_status
+      # - run:
+      #     name: Linting
+      #     command: golint -set_exit_status
       - run:
           name: Build
           command: go build
       - run:
           name: Test
           command: go test -v -race ./...
-    - run:
-        name: Login to Docker
-        command: docker login -u "$DOCKER_USER" -p "$DOCKER_PASSWORD"
-    - run:
-        name: Start QIX Engine
-        command: ACCEPT_EULA=yes docker-compose -f test/docker-compose.yml up -d
-    - run:
-        name: Check that QIX Engine is up and running
-        command: |
-            set +e
-            # Check that Engine is running
-            ENGINE_ID=$(docker ps -aqf "name=engine")
-            echo "Engine container id is $ENGINE_ID"
-            RETRIES=0
-            while [[ "$ENGINE_STATUS" != "running" && $RETRIES -le 30 ]]; do
-                ENGINE_STATUS=$(docker inspect -f '{{.State.Status}}' "$ENGINE_ID")
-                echo "Engine status is $ENGINE_STATUS"
-                sleep 2
-                RETRIES=$[$RETRIES+1]
-            done
-            if [[ "$ENGINE_STATUS" != "running"  ]]; then
-                echo "QIX Engine did not start up properly"
-                exit 1
-            fi
-    - run:
-        name: Integration Test
-        command: ./test/test.sh
\ No newline at end of file
+      - setup_remote_docker
+      - run:
+          name: Run Integration Test
+          working_directory: /go
+          command: |
+            # Spin up environment
+            ACCEPT_EULA=yes docker-compose -f src/github.com/qlik-oss/core-cli/test/docker-compose.yml up -d
+            CONTAINER_ID=$(docker ps -aqf "name=engine")
+            TEST_HOST=$(docker inspect -f '{{range .NetworkSettings.Networks}}{{.Gateway}}{{end}}' "$CONTAINER_ID")
+            # Start a container for test execution
+            TEST_CONTAINER=$(docker run -e TEST_HOST=$TEST_HOST -d golang:1.9-alpine tail -f /dev/null)
+            # Install packages
+            docker exec $TEST_CONTAINER sh -c 'apk update && apk add --no-cache bash make'
+            # Copy code into container
+            docker cp . $TEST_CONTAINER:/go
+            # Copy apps and data into engine container
+            chmod -R 777 src/github.com/qlik-oss/core-cli/test/
+            docker cp src/github.com/qlik-oss/core-cli/test/apps/ $CONTAINER_ID:/
+            docker cp src/github.com/qlik-oss/core-cli/test/data/ $CONTAINER_ID:/
+            # Execute tests
+            docker exec $TEST_CONTAINER /bin/bash -c 'go test /go/src/github.com/qlik-oss/core-cli/test/cli_integration_test.go --engineIP $TEST_HOST:9076'
+            # Bring down the docker-compose and test container
+            docker-compose -f src/github.com/qlik-oss/core-cli/test/docker-compose.yml down
```

Comment:
You shouldn't need to have to clean up, CCI should handle that for you

Reply:
Sweet! Removing ASAP :)